cpp/src/io/parquet/page_enc.cu

```diff
@@ -2017,9 +2022,10 @@ void InitPageFragments(device_2dspan<PageFragment> frag,
                        uint32_t fragment_size,
                        rmm::cuda_stream_view stream)
 {
-  auto num_columns              = frag.size().first;
-  auto num_fragments_per_column = frag.size().second;
-  dim3 dim_grid(num_columns, num_fragments_per_column);  // 1 threadblock per fragment
+  int const num_columns              = frag.size().first;
+  int const num_fragments_per_column = frag.size().second;
+  auto const grid_y                  = std::min(num_fragments_per_column, (1 << 16) - 1);
```

Comment:
Does `1 << 16` represent 128 * 512?

Reply:
yes, or however you want to represent 64k 😄